### ISSUE TITLE
Hooks validation does not match docs and validation message has a broken link

### ISSUE THREAD

**Issue by bengourley** [expo/expo-cli#333]
## Description

The "latest" documentation for [Hooks](https://docs.expo.io/versions/latest/workflow/configuration/#hooks) says that "postPublish" should be a `String` but the CLI expects an array. The resulting error message includes a broken link.

## Expected Behavior

I provided the following in my `app.json`:

```json
{
  "hooks": {
    "postPublish": "echo HELLO"
  }
}
```

I would have expected this to either "just work" or fail with a reason as to why. (aside: the documentation doesn't tell me if it will execute the hook string as a shell command, or I should provide a JS module name for it to load?)

## Observed Behavior

```
Development/my-expo-project - [master●] » expo start
[13:37:01] Starting project at /Users/bengourley/Development/my-expo-project
[13:37:01] Expo DevTools is running at http://localhost:19002
[13:37:01] Opening DevTools in the browser... (press shift-d to disable)
[13:37:03] Error: Problem validating fields in app.json. See https://docs.expo.io/versions/v32.0.0/guides/configuration.html
[13:37:03]  • Field: hooks.postPublish - should be array.
```

The link "https://docs.expo.io/versions/v32.0.0/guides/configuration.html" does not work.

## Environment

```
  Expo CLI 2.7.0 environment info:
    System:
      OS: macOS Sierra 10.12.6
      Shell: 5.2 - /bin/zsh
    Binaries:
      Node: 10.4.1 - /usr/local/bin/node
      Yarn: 1.12.3 - ~/.yarn/bin/yarn
      npm: 6.1.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.1 AI-173.4720617
      Xcode: 9.2/9C40b - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.0 => 32.0.2
      react: 16.5.0 => 16.5.0
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1
    npmGlobalPackages:
      expo-cli: 2.7.0
```

## Reproducible Demo

_[I don't think I need to produce one but let me know]_

**Comment by fson** [comment]
Hey @bengourley! Sorry that the documentation for `postPublish` hooks is lacking at the moment.

Unfortunately the field type "string" in `app.json` documentation is incorrect, you need to set `postPublish` to an array of objects like this:
```
{ "file": "./path/to-a-module.js", "config": {} }
```
(See the Sentry guide for a working example of this: https://docs.expo.io/versions/latest/guides/using-sentry/#install-and-configure-sentry)

The `file` field should be a JS module that exports a function. The function is called with an object shaped like this, after publish:
```
{
  config,       // the config object provided in app.json
  url,          // the URL of the published app
  exp,          // the full app.json configuration
  iosBundle,
  iosSourceMap,
  iosManifest,
  androidBundle,
  androidSourceMap,
  androidManifest,
  projectRoot,
  log,           // a logger function
}
```

**Comment by bengourley** [comment]
Thanks! I figured that out by reading the source and got something working.

_Aside: I work at @bugsnag so our use case is very much similar – uploading source maps to our API when an app is built._

I think the broken link needs fixing in this package as well as the docs being updated?

**Comment by fson** [comment]
@AdamJNavarro could you make sure postPublish hooks are correctly documented at https://docs.expo.io/versions/latest/workflow/configuration/#hooks?

**Comment by EvanBacon** [comment]
We now maintain TS types for the ExpoConfig, here is [postPublish hooks](https://github.com/expo/expo-cli/blob/dc82a5de9e6cbc155ec410f28af7f14eb4e1d7d2/packages/config-types/src/ExpoConfig.ts#L248). Docs can be found here [hooks.postPublish](https://docs.expo.io/versions/latest/config/app/#postpublish)

**Comment by EvanBacon** [comment]
We've also fixed the log link in 137bfae
